### PR TITLE
Hyc 767 1science ingest

### PR DESCRIPTION
**Pull request by smithjp**
* https://jira.lib.unc.edu/browse/HYC-767

### CODE REVIEW

File: lib/tasks/onescience/onescience_ingest.rake [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+          work_attributes[k] = v.first
+        elsif work.attributes.keys.member?(k.to_s) && work.attributes[k.to_s].respond_to?(:each) && !work_attributes[k].respond_to?(:each)
+          work_attributes[k] = Array(v)
+        else
+          work_attributes[k] = v
+        end
+      end
+
+      # Only keep attributes which apply to the given work type
+      work.attributes = work_attributes.reject{|k,v| !work.attributes.keys.member?(k.to_s) unless k.to_s.ends_with? '_attributes'}
```
Comment (bbpennel) [inline lib/tasks/onescience/onescience_ingest.rake]: Log errors

File: lib/tasks/onescience/onescience_ingest.rake [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    work_attributes['issn'] = onescience_data['ISSNs'].split('||') if !onescience_data['ISSNs'].blank?
+    work_attributes['abstract'] = onescience_data['Abstract']
+    work_attributes['keyword'] = onescience_data['Keywords'].split('||') if !onescience_data['Keywords'].blank?
+    work_attributes['creators_attributes'] = get_people(onescience_data['onescience_id'])
+    work_attributes['resource_type'] = 'Article'
+    work_attributes['language'] = 'http://id.loc.gov/vocabulary/iso639-2/eng'
+    work_attributes['dcmi_type'] = 'http://purl.org/dc/dcmitype/Text'
+    work_attributes['admin_set_id'] = @admin_set_id
+    # edition?
+    # rights statement?
```
Comment (bbpennel) [inline lib/tasks/onescience/onescience_ingest.rake]: Check with anna

File: lib/tasks/onescience/onescience_ingest.rake [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        puts "[#{Time.now}] #{item_data['onescience_id']},#{work.id} attaching files"
+        sources = []
+        file_count = files.count
+
+        # Move pubmed file to beginning of hash so it will be the primary work
+        if files.key?('PubMedCentral-Link_Files')
+          files = {'PubMedCentral-Link_Files' => files['PubMedCentral-Link_Files']}.merge(files)
+        end
+
+        files.each_with_index do |(k,v),file_index|
```
Comment (bbpennel) [inline lib/tasks/onescience/onescience_ingest.rake]: Use more informative variable names

File: lib/tasks/onescience/onescience_ingest.rake [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    work_attributes['admin_set_id'] = @admin_set_id
+    # edition?
+    # rights statement?
+    files = onescience_data.select { |k,v| k['Files'] && !v.blank? }
+
+    [work_attributes, files]
+  end
+
+  def get_people(onescience_id)
+    people = {}
```
Comment (bbpennel) [inline lib/tasks/onescience/onescience_ingest.rake]: Maybe this should be an array?

Comment (smithjp) [reply lib/tasks/onescience/onescience_ingest.rake]: the structure for the people objects is 
`"advisors_attributes"=>{"0"=>{"name"=>"a name", "affiliation"=>"unc affiliation", "other_affiliation"=>"non-unc affiliation", "orcid"=>"an orchid id"},"1"=>{"name"=>"", "affiliation"=>"", "other_affiliation"=>"", "orcid"=>""}}`
so i think the `{}` need to stay

File: lib/tasks/onescience/onescience_ingest.rake (hunk 1 above)
Comment (smithjp) [reply lib/tasks/onescience/onescience_ingest.rake]: added log below 121-132

**Pull request merged**